src/text-area/text-area.stories.mdx

```diff
@@ -182,20 +183,29 @@ Note that these variables are shared with other components such as `Textfield`,
 </Canvas>
 
 export function AutoExpandStory(props) {
+    const [value, setValue] = React.useState('')
     return (
-        <Stack space="xxlarge" dividers="secondary" maxWidth="medium">
+        <Stack space="large" dividers="secondary" maxWidth="medium">
             <TextArea
-                {...props}
-                label="What do you want to accomplish?"
-                message="Write as much or as little as you want. The input area will auto-expand to fit what you've typed."
+                label="Text area with auto-expand"
+                auxiliaryLabel="(controlled)"
                 autoExpand
+                value={value}
+                onChange={(event) => setValue(event.target.value)}
+                message="Write as much or as little as you want. The input area will auto-expand to fit what you've typed."
+                rows={1}
+                onKeyDown={(event) => {
+                    if (event.key === 'Enter') {
+                        event.preventDefault()
+                        setValue('') // Clear the input programmatically
+                    }
+                }}
             />
-            <TextArea
-                {...props}
-                label="What do you want to accomplish?"
-                message="This one will not auto-expand."
-                autoExpand={false}
```

Comment:
💭 Was it intentional to remove the "What do you want to accomplish?" textarea that had `autoExpand={false}`? I'm ok without it (as anyway there are plenty of other stories with textareas with `autoExpand={false}`